app/src/main/java/ch/epfl/sdp/social/MessageAdapter.java

```diff
@@ -71,6 +71,17 @@ public View getView(int i, View convertView, ViewGroup viewGroup) {
         return convertView;
     }
 
+    public void sortMessages() {
+        for (int i = 0; i < messages.size()-1; i++)
+            for (int j = 0; j < messages.size()-i-1; j++)
+                if (messages.get(j).getM().getDate().compareTo(messages.get(j+1).getM().getDate())>0)
+                {
+                    ChatActivity.MessageDecorator temp = messages.get(j);
+                    messages.set(j, messages.get(j+1));
+                    messages.set(j+1,temp);
+                }
+    }

```

Comment:
This is correct sorting however the way of doing this I imagined was completely different (and much more efficient). First, you begin by the assumption that the messages list inside messageAdapter is sorted. This initial condition is satisfied since initially the messages list is empty and hence sorted. Once you call add (inside MessageAdapter) you just have to use the loop invariant that the list is already sorted and find the correct position for you to insert the new MessageDecorator object. To reduce the complexity of insertion, you can change messages list from ArrayList to LinkedList (which is a doubly-linked list).  Please let me know if you need more explanation regarding this.

Reply:
I get the idea, but the dev note only mentionned bubble sort.